### ISSUE TITLE
TextInput inside TapGestureHandler cancels tap events.

### ISSUE THREAD

**Issue by vpuonti** [software-mansion/react-native-gesture-handler#774]
# Description

As stated in the title, having a TextInput inside a TapGestureHandler cancels the tap events on Android. 

This sucks because at the moment, TextInput is the only way (that I know of) to display an Animated value in the UI. I'm attempting to create a UI where clicking the TextInput "selects" the value for editing, but that's not possible as I'm not getting the events from the handler.

# Snack

https://snack.expo.io/@valtteri.puonti-cgi/842ba9

# Relevant code

```jsx
const {event, call, Value} = Animated

export default class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      textInputRef: React.createRef(),
      tapStateText: '',
      tapOldStateText: ''
    }
  }

  render() {
    return (
    <View style={styles.container}>
      <TapGestureHandler
        onHandlerStateChange={event([{
          nativeEvent: {
            state: state => call([state], (args) => {
              this.setState({tapStateText: `tapState: ${args[0]}`})
            }),
            oldState: oldState => call([oldState], (args) => {
              this.setState({tapOldStateText: `tapOldState: ${args[0]}`})
            })
          }
        }])}
      >
        <Animated.View style={{ flex: 1 }}>
          <TextInput
            ref={this.state.textInputRef}
            style={{ flex: 1, borderWidth: StyleSheet.hairlineWidth }}
            editable={false}
          >
          {this.state.tapStateText} - {this.state.tapOldStateText}
          </TextInput>
        </Animated.View>
      </TapGestureHandler>
    </View>

    );
  }
}

```

**Comment by vpuonti** [comment]
I got the same functionality I looked for by putting the TapGestureHandler as a sibling of the TextInput, and giving the TapGestureHandler's child View styles so it fills the parent view and stays in front of the TextInput.

# Code:

```jsx
<Animated.View
  style={{
    flex: 1,
  }}
>
  <TextInput
    ref={this.selectionStartDisplayRef}
    editable={false}
  />
  <TapGestureHandler
    onHandlerStateChange={e => console.log(e.nativeEvent.state)}
  >
    <Animated.View
      style={{
        zIndex: 1,
        position: 'absolute',
        ...StyleSheet.absoluteFillObject,
      }}
    />
  </TapGestureHandler>
</Animated.View>
```

**Comment by skylarkgit** [comment]
+1 I am facing the same issue. There should be a neat solution to propagate the event up.

**Comment by jkadamczyk** [comment]
The repro is for old version of snack so needs to be recreated. This shouldn't be a huge issue.

**Comment by yousseftarekkh** [comment]
[Snack](https://snack.expo.dev/@yousseftarek/react-native-gesture-handler) recreated, issue persists.

**Comment by j-piasecki** [comment]
@yousseftarekkh The combination of https://github.com/software-mansion/react-native-gesture-handler/pull/2016 and https://github.com/software-mansion/react-native-gesture-handler/pull/2017, included in GH 2.5.0, should fix your snack. It will require some changes though:
- you need to import `TextInput` from `react-native-gesture-handler` instead of `react-native`
- add `simultateousHandlers={controller3Ref}` to the third `TextInput`.

There's still (on Android), however, a problem with handling gestures attached to the `TextInput` when its `editable` property is set to `false`.
